src/http-api/routes/pubsub.js

```diff
@@ -1,38 +1,23 @@
 'use strict'
 
-const Joi = require('joi')
 const resources = require('./../resources')
 
 module.exports = (server) => {
   const api = server.select('API')
 
   api.route({
     method: '*',
-    path: '/api/v0/pubsub/sub/{topic}',
+    path: '/api/v0/pubsub/sub',
     config: {
-      handler: resources.pubsub.subscribe.handler,
-      validate: {
-        params: {
-          topic: Joi.string().required()
-        },
-        query: {
-          discover: Joi.boolean()
-        }
-      }
+      handler: resources.pubsub.subscribe.handler
```

Comment:
so the changes are two fold, 1) match the format that go-ipfs uses for these urls and 2) drop joi validation, because I can't seem to make it work with all the options, which is unfortunate but something I need to look into for all of the http-api not just this part.